airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/read/SelectQuerier.kt

```diff
@@ -105,31 +123,34 @@ class JdbcSelectQuerier(
             return hasNext
         }
 
-        override fun next(): ObjectNode {
-            metaChanges.clear()
+        override fun next(): SelectQuerier.ResultRow {
             // Ensure that the current row in the ResultSet hasn't been read yet; advance if
             // necessary.
             if (!hasNext()) throw NoSuchElementException()
             // Read the current row in the ResultSet
-            val record: ObjectNode = reusable ?: Jsons.objectNode()
+            val resultRow: ResultRow = reusable ?: ResultRow()
+            resultRow.changes.clear()
             var colIdx = 1
             for (column in q.columns) {
                 log.debug { "Getting value #$colIdx for $column." }
                 val jdbcFieldType: JdbcFieldType<*> = column.type as JdbcFieldType<*>
                 try {
-                    record.set<JsonNode>(column.id, jdbcFieldType.get(rs!!, colIdx))
+                    resultRow.data.set<JsonNode>(column.id, jdbcFieldType.get(rs!!, colIdx))
                 } catch (e: Exception) {
-                    record.set<JsonNode>(column.id, Jsons.nullNode())
-                    log.info {
-                        "Failed to serialize column: ${column.id}, of type ${column.type}, with error ${e.message}"
+                    resultRow.data.set<JsonNode>(column.id, Jsons.nullNode())
+                    if (!hasLoggedException) {
+                        log.warn(e) { "Error deserializing value in column $column." }
+                        hasLoggedException = true
+                    } else {
+                        log.debug(e) { "Error deserializing value in column $column." }
```

Comment:
I thought it would be cute to log the first error in the results as WARN and the rest as DEBUG

Reply:
I don't know about cute, but it makes sense